pandas/core/arrays/arrow/array.py

```diff
@@ -2523,7 +2526,15 @@ def _str_get_dummies(self, sep: str = "|"):
         n_cols = len(uniques)
         indices = pc.index_in(flattened_values, uniques_sorted).to_numpy()
         indices = indices + np.arange(n_rows).repeat(lengths) * n_cols
-        dummies = np.zeros(n_rows * n_cols, dtype=np.bool_)
+        _dtype = pandas_dtype(dtype)
+        dummies_dtype: NpDtype
+        if isinstance(_dtype, np.dtype):
+            dummies_dtype = _dtype
+        else:
+            dummies_dtype = np.bool_
+        dummies = np.zeros(n_rows * n_cols, dtype=dummies_dtype)
+        if dtype == str:
+            dummies[:] = False
```

Comment:
Can you just put this logic before the `dummies` creation i.e.

```python
if dtype == str:
    dummies_dtype = np.bool_
dummies = ...
```

Reply:
The string types do not need to use a dummy dtype, they can handle boolean values. The issue is with the str type interaction with np.zeroes(), where it considers the zero value to be the empty string.